daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilInputContentHandler.scala

```diff
@@ -0,0 +1,180 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.processors
+
+import scala.util.Try
+import scala.xml.NamespaceBinding
+
+import javax.xml.XMLConstants
+import org.apache.daffodil.api.DFDL
+import org.apache.daffodil.api.DFDL.DaffodilUnhandledSAXException
+import org.apache.daffodil.api.DFDL.DaffodilUnparseErrorSAXException
+import org.apache.daffodil.infoset.IllegalContentWhereEventExpected
+import org.apache.daffodil.infoset.InfosetInputterEventType.EndDocument
+import org.apache.daffodil.infoset.InfosetInputterEventType.EndElement
+import org.apache.daffodil.infoset.InfosetInputterEventType.StartDocument
+import org.apache.daffodil.infoset.InfosetInputterEventType.StartElement
+import org.apache.daffodil.infoset.SAXInfosetInputter
+import org.apache.daffodil.util.Maybe.Nope
+import org.apache.daffodil.util.Maybe.One
+import org.xml.sax.Attributes
+import org.xml.sax.Locator
+
+class DaffodilInputContentHandler(dp: DFDL.DataProcessor, output: DFDL.Output)
+  extends DFDL.DaffodilInputContentHandler {
+  private val inputter = new SAXInfosetInputter(this, dp, output)
+  private var unparseResult: DFDL.UnparseResult = _
+  private var infosetEvent: DFDL.SaxInfosetEvent = _
+  private val characterData = new StringBuilder()
+
+  def setUnparseResult(ur: DFDL.UnparseResult): Unit = {
+    unparseResult = ur
+  }
+
+  def getUnparseResult: DFDL.UnparseResult = unparseResult
+
+  override def setDocumentLocator(locator: Locator): Unit = {
+    // do nothing
+  }
+
+  override def startDocument(): Unit = {
+    infosetEvent = new DFDL.SaxInfosetEvent
+    infosetEvent.eventType = One(StartDocument)
+    sendToInputter
+  }
+
+  override def endDocument(): Unit = {
+    if (infosetEvent == null) {
+      infosetEvent = new DFDL.SaxInfosetEvent
+    }
+    infosetEvent.eventType = One(EndDocument)
+    sendToInputter
+  }
+
+  // TODO not sure what to do with the prefix mappings, technically we don't need them since
+  // we have the attributes with startElement to look up any prefixes we need
+  override def startPrefixMapping(prefix: String, uri: String): Unit = {
+    val pre = if (prefix == "") null else prefix
+    infosetEvent.prefixMapping = One(NamespaceBinding(pre, uri, infosetEvent.prefixMapping
+      .orNull))
+  }
```

Comment:
Yeah, seem like we don't need it. Unless there are XMLReaders don't prvide certain parametrs in startELement? LIke maybe uri is always null, and they expect you to figure out the namespace uri based on the qName string?

Reply:
The expectation is for the readers to support the features, which a user can set to either get the uri and localName or just the qualified qName. So I think we can build this mapping and use it together with the Attributes in startElement to ensure we all the info to get a URI